MeshLib/MeshSubset.h

```diff
@@ -22,61 +22,38 @@
 
 namespace MeshLib
 {
-
-/// A subset of nodes or elements on a single mesh.
-class MeshSubset
+inline std::vector<Node*> nodesNodesIntersection(
+    std::vector<Node*> const& nodes_a, std::vector<Node*> const& nodes_b)
 {
-public:
-    /// Construct a mesh subset from vector of nodes on the given mesh.
-    /// \param msh Mesh
-    /// \param vec_items Vector of Node pointers.
-    /// \param delete_ptr Deletes the vector of Node pointers if true.
-    /// \note When delete_ptr is set only the vector is deleted, not the
-    /// elements of the vector.
-    MeshSubset(const Mesh& msh, std::vector<Node*> const* vec_items,
-        bool const delete_ptr = false)
-        : _msh(msh), _nodes(vec_items), _eles(nullptr), _delete_ptr(delete_ptr)
-    {}
-
-    /// Construct a mesh subset from vector of elements on the given mesh.
-    /// \param msh Mesh
-    /// \param vec_items Vector of Element pointers.
-    /// \param delete_ptr Deletes the vector of Element pointers if true.
-    /// \note When delete_ptr is set only the vector is deleted, not the
-    /// elements of the vector.
-    MeshSubset(const Mesh& msh, std::vector<Element*> const* vec_items,
-        bool const delete_ptr = false)
-        : _msh(msh), _nodes(nullptr), _eles(vec_items), _delete_ptr(delete_ptr)
-    {}
+    if (nodes_a.empty() || nodes_b.empty())
+    {
+        return {};
+    }
 
-    /// construct from both nodes and elements
-    /// Construct a mesh subset from vector of nodes and a vector of elements on
-    /// the given mesh.
-    /// \param msh Mesh
-    /// \param vec_nodes Vector of Node pointers.
-    /// \param vec_eles Vector of Element pointers.
-    /// \param delete_ptr Deletes the vector of Node pointers if true.
-    /// \note When delete_ptr is set only the vectors are deleted, not the
-    /// elements of the vectors.
-    MeshSubset(const Mesh& msh, std::vector<Node*> const* vec_nodes,
-        std::vector<Element*> const* vec_eles, bool const delete_ptr = false)
-        : _msh(msh), _nodes(vec_nodes), _eles(vec_eles), _delete_ptr(delete_ptr)
-    {}
+    std::vector<Node*> active_nodes;
 
-    ~MeshSubset()
+    for (auto const& n_a : nodes_a)
     {
-        if (_delete_ptr)
+        auto it = std::find(begin(nodes_b), end(nodes_b), n_a);
+        if (it != end(nodes_b))
         {
-            delete _nodes;
-            delete _eles;
+            active_nodes.push_back(n_a);
         }
     }
 
-    /// return the total number of mesh items
-    std::size_t getNumberOfTotalItems() const
-    {
-        return getNumberOfNodes() + getNumberOfElements();
-    }
+    return active_nodes;
+}
+
+/// A subset of nodes on a single mesh.
```

Comment:
We may use elements as a MeshSubset to perform deactivate/activating elements for different processes in a coupling or for excavation modelling. If only nodes are kept in this class, the class name, I think, it can be termed as  MeshNodeSubset. Then if we need the functionality  of deactivate/activating elements, we can introduce a class of  MeshElementSubset.

Reply:
I agree with renaming in general. I'd postpone it until we have the hypothetical MeshElementSubset because the rename touches 43 files.